src/algebra/hom/units.lean

```diff
@@ -103,31 +108,162 @@ def to_hom_units {G M : Type*} [group G] [monoid M] (f : G →* M) : G →* Mˣ
 
 end monoid_hom
 
-section is_unit
-variables {M : Type*} {N : Type*}
+namespace is_unit
+variables {F α M N : Type*}
+
+section monoid
+variables [monoid M] [monoid N]
 
-@[to_additive] lemma is_unit.map {F : Type*} [monoid M] [monoid N] [monoid_hom_class F M N]
-  (f : F) {x : M} (h : is_unit x) : is_unit (f x) :=
+@[to_additive] lemma map [monoid_hom_class F M N] (f : F) {x : M} (h : is_unit x) : is_unit (f x) :=
 by rcases h with ⟨y, rfl⟩; exact (units.map (f : M →* N) y).is_unit
 
 /-- If a homomorphism `f : M →* N` sends each element to an `is_unit`, then it can be lifted
 to `f : M →* Nˣ`. See also `units.lift_right` for a computable version. -/
 @[to_additive "If a homomorphism `f : M →+ N` sends each element to an `is_add_unit`, then it can be
 lifted to `f : M →+ add_units N`. See also `add_units.lift_right` for a computable version."]
-noncomputable def is_unit.lift_right [monoid M] [monoid N] (f : M →* N)
-  (hf : ∀ x, is_unit (f x)) : M →* Nˣ :=
+noncomputable def lift_right (f : M →* N) (hf : ∀ x, is_unit (f x)) : M →* Nˣ :=
 units.lift_right f (λ x, (hf x).unit) $ λ x, rfl
 
-@[to_additive] lemma is_unit.coe_lift_right [monoid M] [monoid N] (f : M →* N)
-  (hf : ∀ x, is_unit (f x)) (x) :
+@[to_additive] lemma coe_lift_right (f : M →* N) (hf : ∀ x, is_unit (f x)) (x) :
   (is_unit.lift_right f hf x : N) = f x := rfl
 
-@[simp, to_additive] lemma is_unit.mul_lift_right_inv [monoid M] [monoid N] (f : M →* N)
-  (h : ∀ x, is_unit (f x)) (x) : f x * ↑(is_unit.lift_right f h x)⁻¹ = 1 :=
+@[simp, to_additive] lemma mul_lift_right_inv (f : M →* N) (h : ∀ x, is_unit (f x)) (x) :
+  f x * ↑(is_unit.lift_right f h x)⁻¹ = 1 :=
 units.mul_lift_right_inv (λ y, rfl) x
 
-@[simp, to_additive] lemma is_unit.lift_right_inv_mul [monoid M] [monoid N] (f : M →* N)
-  (h : ∀ x, is_unit (f x)) (x) : ↑(is_unit.lift_right f h x)⁻¹ * f x = 1 :=
+@[simp, to_additive] lemma lift_right_inv_mul (f : M →* N) (h : ∀ x, is_unit (f x)) (x) :
+  ↑(is_unit.lift_right f h x)⁻¹ * f x = 1 :=
 units.lift_right_inv_mul (λ y, rfl) x
 
+end monoid
+
+section division_monoid
+variables [division_monoid α] {a b c : α}
+
+@[simp, to_additive] protected lemma inv_mul_cancel : is_unit a → a⁻¹ * a = 1 :=
+by { rintro ⟨u, rfl⟩, rw [←units.coe_inv, units.inv_mul] }
+
+@[simp, to_additive] protected lemma mul_inv_cancel : is_unit a → a * a⁻¹ = 1 :=
+by { rintro ⟨u, rfl⟩, rw [←units.coe_inv, units.mul_inv] }
+
+/-- The element of the group of units, corresponding to an element of a monoid which is a unit. -/
+@[to_additive "The element of the additive group of additive units, corresponding to an element of
+an additive monoid which is an additive unit.", simps]
+def unit' (h : is_unit a) : αˣ := ⟨a, a⁻¹, h.mul_inv_cancel, h.inv_mul_cancel⟩
```

Comment:
What's the difference between this and `is_unit.unit`?

Reply:
Oh, it's the computable one with a definitionally better inverse! Can you make that clear in the docstring, and add a lemma stating that they're equal?

Also, why is this

Reply:
Linking with [to_units](https://leanprover-community.github.io/mathlib_docs/algebra/hom/equiv.html#to_units) would also be good.

Reply:
Can you think of a better naming scheme than `is_unit.unit` and `is_unit.unit'`? It makes it seem like `is_unit.unit'` is less interesting somehow.

Reply:
> Linking with [to_units](https://leanprover-community.github.io/mathlib_docs/algebra/hom/equiv.html#to_units) in the docstring would also be good.

Do you mean cross ref in the docstrings, or redefine using it?

Reply:
A crossref is fine